File: kinara_lms/overrides/company.py
import frappe

def validate_default_disbursement_account(doc, method=None):
    if doc.default_disbursement_account:
        account = frappe.get_doc("Bank Account", doc.default_disbursement_account)
        if account.company != doc.company_name:
            frappe.throw(f"Account Should Belong to the Same Company")

File: kinara_lms/public/js/erpnext/company.js
frappe.ui.form.on("Company", {
    refresh(frm){
        defaultDisbursementAccountFilter(frm);
    },
});

function defaultDisbursementAccountFilter(frm){
    
    frm.set_query('default_disbursement_account', () => {
        return {
            filters: [
                ["company", "=", frm.doc.company_name],
            ]
        };
    });
}